api/src/main/java/org/pmiops/workbench/mailchimp/MailChimpServiceImpl.java

```diff
@@ -40,6 +40,9 @@ public MailChimpServiceImpl(Provider<CloudStorageService> cloudStorageServicePro
   @Override
   public String addUserContactEmail(String contactEmail) {
     String userId;
+    if (listId == null) {
```

Comment:
Wouldn't apiKey also be a problem here? Perhaps private getters for listId and apiKey which cache the values, otherwise you'll be doing this in every method.

Separately, do we need to worry about concurrent access to this service?

Reply:
The API key is also fetched as a part of the call. But here we needed to construct an object using the list ID before we actually call getClient.